store/sqlstore/channel_store.go

```diff
@@ -1755,13 +1755,63 @@ func (s SqlChannelStore) AutocompleteInTeamForSearch(teamId string, userId strin
 			}
 		}
 
+		directChannels, err := s.autocompleteInTeamForSearchDirectMessages(userId, term)
+		if err != nil {
+			result.Err = err
+			return
+		}
+
+		channels = append(channels, directChannels...)
+
 		sort.Slice(channels, func(a, b int) bool {
 			return strings.ToLower(channels[a].DisplayName) < strings.ToLower(channels[b].DisplayName)
 		})
 		result.Data = &channels
 	})
 }
 
+func (s SqlChannelStore) autocompleteInTeamForSearchDirectMessages(userId string, term string) ([]*model.Channel, *model.AppError) {
+	queryFormat := `
+			SELECT
+				C.*,
+				OtherUsers.Username as DisplayName
```

Comment:
Isn't it up to the client to decide if it wants to render the `username` or the `FirstName LastName`?

Reply:
This is for the autocompletion of the "in:" search filter, which use channel names, so I exposed here usernames because is the most equivalent for users. Actually the username is what is used for the "filter" itself, so when you write part of the filter, the autocompletion list shows you the usernames, and if you click in one username, it will set in:@username, so... in this case makes a lot of sense to show always the username, instead the First/Last name